include/llvm/Support/WinTypes.h

```diff
@@ -42,45 +41,51 @@ typedef DWORD *LPDWORD;
 typedef uint32_t UINT32;
 typedef uint64_t UINT64;
 
-typedef signed char         INT8, *PINT8;
-typedef signed int          INT32, *PINT32;
-
+typedef signed char INT8, *PINT8;
+typedef signed int INT32, *PINT32;
 
 typedef size_t SIZE_T;
-typedef const char* LPCSTR;
-typedef const char* PCSTR;
+typedef const char *LPCSTR;
+typedef const char *PCSTR;
 
 typedef int errno_t;
 
 typedef wchar_t WCHAR;
-typedef wchar_t* LPWSTR;
-typedef wchar_t* PWCHAR;
-typedef const wchar_t* LPCWSTR;
-typedef const wchar_t* PCWSTR;
+typedef wchar_t *LPWSTR;
+typedef wchar_t *PWCHAR;
+typedef const wchar_t *LPCWSTR;
+typedef const wchar_t *PCWSTR;
 
 typedef WCHAR OLECHAR;
 typedef OLECHAR *BSTR;
 typedef OLECHAR *LPOLESTR;
-typedef char* LPSTR;
+typedef char *LPSTR;
 
-typedef void* LPVOID;
+typedef void *LPVOID;
 typedef const void *LPCVOID;
 
 typedef std::nullptr_t nullptr_t;
 
-typedef struct _GUID {
-  DWORD Data1;
-  WORD  Data2;
-  WORD  Data3;
-  BYTE  Data4[8];
-} GUID;
+struct GUID {
+  uint32_t Data1;
+  uint16_t Data2;
+  uint16_t Data3;
+  uint8_t Data4[8];
+};
 typedef GUID CLSID;
-typedef const GUID& REFGUID;
-typedef std::type_index REFIID;
-typedef const GUID& REFCLSID;
+typedef const GUID &REFGUID;
+typedef const void *REFIID;
+typedef const GUID &REFCLSID;
 
-#define __uuidof(T) std::type_index(typeid(T))
-#define IID_PPV_ARGS(ppType) __uuidof(**(ppType)), reinterpret_cast<void**>(ppType)
+// The following macros are defined to facilitate the lack of 'uuid' on Linux.
+#define DECLARE_CROSS_PLATFORM_UUIDOF(T)                                       \
+private:                                                                       \
+  static const char T##_ID;                                                    \
+public:                                                                        \
+  static REFIID uuidof() { return static_cast<REFIID>(&T##_ID); }
```

Comment:
I suppose a macro switching the public/private state is fine provided it's the last in the class. I feel like there might be a way to guarantee that, but I can't think of anything. Might bear mentioning in a comment, but might not be worth resubmitting for.

Reply:
I'm trying to come up with another solution maybe a member struct that has a private static member, but it's not working out so far.

Reply:
I will fix by putting `public:` first and then the `private:` section.
@pow2clk Feel free to let us know how it works out. We want to use the best solution possible.